sql/testing/path_test.cc

```diff
@@ -231,10 +231,10 @@ BOOST_AUTO_TEST_CASE(test_indexes)
 {
     BOOST_CHECK_EQUAL(PathElement(0).toIndex(), 0);
     BOOST_CHECK_EQUAL(PathElement("0").toIndex(), 0);
-    BOOST_CHECK_EQUAL(PathElement("00").toIndex(), 0);
+    BOOST_CHECK_EQUAL(PathElement("00").toIndex(), -1);
     BOOST_CHECK_EQUAL(PathElement(123456789).toIndex(), 123456789);
     BOOST_CHECK_EQUAL(PathElement("123456789").toIndex(), 123456789);
-    BOOST_CHECK_EQUAL(PathElement("0123456789").toIndex(), 123456789);
+    BOOST_CHECK_EQUAL(PathElement("0123456789").toIndex(), -1);
     BOOST_CHECK_EQUAL(PathElement(-1).toIndex(), -1);
     BOOST_CHECK_EQUAL(PathElement(-1000).toIndex(), -1);
```

Comment:
BOOST_CHECK_EQUAL(PathElement("123456789").toIndex(), 123456789);

That tests the positive side.